### PR TITLE
BUGFIX: TMA-1218 Anonymise hidden parameters only in LCM bricks running in K8s

### PR DESCRIPTION
**Pull request by sadam21**
Also remove duplicate parameters 'process_id' and 'project_id' and rename
'execution_id' parameter to 'GDC_EXECUTION_ID' to be consistent with other
GDC parameters.

### CODE REVIEW

File: bin/run_brick.rb [hunk 1]
```diff
@@ -13,17 +14,22 @@
 
 log.info "action=#{brick_type}_execution status=init"
 
+def get_brick_params(prefix)
+  ENV.select {|k,| k.to_s.match(/^#{prefix}.*/)}.map {|k, v| [k.slice(prefix.length..-1), v]}.to_h
```
Comment (yenkins) [inline bin/run_brick.rb]: info: Has the variable name 'k' (UncommunicativeVariableName) (reek)

Comment (yenkins) [inline bin/run_brick.rb]: info: Has the variable name 'v' (UncommunicativeVariableName) (reek)

Comment (yenkins) [inline bin/run_brick.rb]: warning: Layout/SpaceInsideBlockBraces: Space between { and | missing. (rubocop)

Comment (yenkins) [inline bin/run_brick.rb]: warning: Layout/SpaceInsideBlockBraces: Space missing inside }. (rubocop)

File: bin/run_brick.rb [hunk 2]
```diff
@@ -13,17 +14,22 @@
 
 log.info "action=#{brick_type}_execution status=init"
 
+def get_brick_params(prefix)
```
Comment (yenkins) [inline bin/run_brick.rb]: info: Doesn't depend on instance state (maybe move it to another class?) (UtilityFunction) (reek)

File: lib/gooddata/bricks/middleware/logger_middleware.rb [hunk 3]
```diff
@@ -28,9 +28,9 @@ def call(params)
         params = params.to_hash
         if params['GDC_LOGGING_OFF']
           logger = NilLogger.new
-        elsif params['log_directory'] && params['execution_id']
-          log_directory = params['log_directory']
-          execution_id = params['execution_id']
+        elsif params['GDC_LOG_DIRECTORY'] && params['GDC_EXECUTION_ID']
```
Comment (yenkins) [inline lib/gooddata/bricks/middleware/logger_middleware.rb]: info: Calls 'params['gdc_execution_id']' 2 times (DuplicateMethodCall) (reek)

Comment (yenkins) [inline lib/gooddata/bricks/middleware/logger_middleware.rb]: info: Calls 'params['gdc_log_directory']' 2 times (DuplicateMethodCall) (reek)

Comment (yenkins) [inline lib/gooddata/bricks/middleware/logger_middleware.rb]: info: Refers to 'params' more than self (maybe move it to another class?) (FeatureEnvy) (reek)

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/917/ : SUCCESS in 9s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1730/ : FAILURE in 1m 21s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1991/ : SUCCESS in 1m 15s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-build-pipeline/168/ : SUCCESS in 7m 43s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/919/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1732/ : SUCCESS in 1m 15s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1993/ : SUCCESS in 1m 10s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-build-pipeline/170/ : SUCCESS in 7m 02s

**Comment by kubamahnert** [comment]
ok to merge

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-merge-compliance-check/155/ : SUCCESS in 29s

**Pull request merged**